pennylane/devices/default_mixed.py

```diff
@@ -903,5 +977,95 @@ def execute(
         self,
         circuits: QuantumScript,
         execution_config: Optional[ExecutionConfig] = None,
-    ) -> None:
-        raise NotImplementedError
+    ) -> Union[Result, ResultBatch]:
+        interface = (
+            execution_config.interface
+            if execution_config.gradient_method in {"best", "backprop", None}
+            else None
+        )
+
+        return tuple(
+            simulate(
+                c,
+                rng=self._rng,
+                prng_key=self._prng_key,
+                debugger=self._debugger,
+                interface=interface,
+                readout_errors=self.readout_err,
+            )
+            for c in circuits
+        )
+
+    def _setup_execution_config(self, execution_config: ExecutionConfig) -> ExecutionConfig:
+        """This is a private helper for ``preprocess`` that sets up the execution config.
+
+        Args:
+            execution_config (ExecutionConfig): an unprocessed execution config.
+
+        Returns:
+            ExecutionConfig: a preprocessed execution config.
+        """
+        updated_values = {}
+        for option in execution_config.device_options:
+            if option not in self._device_options:
+                raise qml.DeviceError(f"device option {option} not present on {self}")
+
+        if execution_config.gradient_method == "best":
+            updated_values["gradient_method"] = "backprop"
+        updated_values["use_device_gradient"] = execution_config.gradient_method in {
+            "backprop",
+            "best",
+        }
+        updated_values["grad_on_execution"] = False
+        updated_values["device_options"] = dict(execution_config.device_options)  # copy
+
+        for option in self._device_options:
+            if option not in updated_values["device_options"]:
+                updated_values["device_options"][option] = getattr(self, f"_{option}")
```

Comment:
can we re-arrange this method to deal with first all the gradient stuff, then all the option stuff (or the other way around)? It just seems easier to parse if we take things one topic at a time, and right now its kind of an `options`-`gradients`-`option` sandwich 🥪 😅